modules/gui/src/com/haulmont/cuba/gui/app/core/entityinspector/EntityInspectorBrowse.java

```diff
@@ -366,16 +366,21 @@ protected void createButtonsPanel(Table table) {
             } catch (FileStorageException e) {
                 log.error("Unable to delete temp file", e);
             }
+            String fileExtension = Files.getFileExtension(importUpload.getFileName());
             try {
                 Collection<Entity> importedEntities;
-                if ("json".equals(Files.getFileExtension(importUpload.getFileName()))) {
+                if ("json".equals(fileExtension)) {
                     importedEntities = entityImportExportService.importEntitiesFromJSON(new String(fileBytes), createEntityImportView(selectedMeta));
                 } else {
                     importedEntities = entityImportExportService.importEntitiesFromZIP(fileBytes, createEntityImportView(selectedMeta));
                 }
                 showNotification(importedEntities.size() + " entities imported", NotificationType.HUMANIZED);
             } catch (Exception e) {
-                showNotification(getMessage("importFailed"), e.getMessage(), NotificationType.ERROR);
+                String message = e.getMessage();
+                if (message != null && !"zip".equals(fileExtension)) {
```

Comment:
Check extension before import, not after the exception. Also, enforce type of file using permittedExtensions property of FileUploadField.